CODING_GUIDELINES.md

```diff
@@ -151,6 +151,55 @@ prior to attempting a search.
 Note: We are in the process of replacing methods that don't adhere to the above principles (for example, return a
 structure instead of a pointer, return a nil error on not-found, etc).
 
+## Implementing functions to support different API versions
+
+Functions dealing with different versions should use a matrix structure to identify which calls to run according to the 
+highest API version supported by vCD. An example can be found in adminvdc.go.
```

Comment:
Knowing how crucial this versioning system is, I'd like to see more explanation how it works & how to use it. We can address other comments and once the details stabilize elaborate this write-up.

Reply:
Please tell more what is missing to make it more elaborated.

Reply:
Actually, with the new var naming it is becoming quite self-explanatory, I'll suggest some improvements in-line.